tests/helpers.py

```diff
@@ -2056,3 +2057,200 @@ def modify_osd_replica_count(resource_name, replica_count):
     params = f'{{"spec": {{"replicas": {replica_count}}}}}'
     resource_name = '-'.join(resource_name.split('-')[0:4])
     return ocp_obj.patch(resource_name=resource_name, params=params)
+
+
+def check_required_osd_count(total_osd_nos=3):
+    """
+    Function to check required OSD count and if count is less add OSDs
+
+    Args:
+        total_osd_nos (int): OSD count per node.
+
+    Returns:
+        cluster_total_osd_count (int): Total OSD count of overall cluster.
+
+    """
+    # Make sure setup has required number of OSDs
+    actual_osd_count = cluster.count_cluster_osd()
+    ocs_nodes = node.get_typed_nodes(node_type='worker')
+    expected_osd_count = len(ocs_nodes) * total_osd_nos
+    if actual_osd_count == expected_osd_count:
+        logging.info(f"Setup has OSD count as per OCS workers")
+    else:
+        storage_cluster.add_capacity(int((expected_osd_count - actual_osd_count) / 3))
```

Comment:
There is no check to see if the OSD count after adding OSDs.

Reply:
add_capacity function has the validation, also it waits for the OSD to be up and running. PR #1550